go-controller/pkg/cni/types.go

```diff
@@ -74,15 +77,25 @@ type PodRequest struct {
 	IfName string
 	// CNI conf obtained from stdin conf
 	CNIConf *types.NetConf
+	// Timestamp when the request was started
+	timestamp time.Time
+	// ctx is a context tracking this request's lifetime
+	ctx context.Context
+	// cancel should be called to cancel this request
+	cancel context.CancelFunc
 }
 
-type cniRequestFunc func(request *PodRequest, kclient kubernetes.Interface) ([]byte, error)
+type cniRequestFunc func(request *PodRequest, podLister corev1listers.PodLister) ([]byte, error)
 
 // Server object that listens for JSON-marshaled Request objects
 // on a private root-only Unix domain socket.
 type Server struct {
 	http.Server
 	requestFunc cniRequestFunc
 	rundir      string
-	kclient     kubernetes.Interface
+	podLister   corev1listers.PodLister
+
+	// adds is a map of sandbox ID to PodRequest for any CNIAdd operation
+	addsLock sync.Mutex
+	adds     map[string]*PodRequest
```

Comment:
nit. naming is hard but this is very confusing

Reply:
maybe pendingCNIAdds ?

Reply:
or cmdAddMap, just that `adds`  and `addsLock` seems is trying to `add` something to